src/main/java/io/confluent/connect/hdfs/wal/FSWAL.java

```diff
@@ -120,6 +121,7 @@ public void apply() throws ConnectException {
           for (Map.Entry<WALEntry, WALEntry> entry: entries.entrySet()) {
             String tempFile = entry.getKey().getName();
             String committedFile = entry.getValue().getName();
+            RecoveryHelper.getInstance().addFile(committedFile);
```

Comment:
I am concerned that the lists in the map grow without bounds and eventually may cause OOM if the process is run long enough.